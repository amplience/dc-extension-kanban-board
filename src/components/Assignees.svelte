<script>
  import type { User } from "dc-extensions-sdk/dist/types/lib/components/Users";
  import Gravatar from "./Gravatar.svelte";

  export let users: User[] = [];

</script>

<style lang="scss">

</style>

<div class="assignees">
  {#each users as user}
  <Gravatar {user} />
  {/each}
</div>
