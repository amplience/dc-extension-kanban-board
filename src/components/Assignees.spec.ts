import { render } from '@testing-library/svelte';
import Assignees from './Assignees.svelte';

jest.mock('md5', () => ({ default: jest.fn().mockReturnValue('HASH') }));

describe('Assignees', () => {
  it('should render a single assignee', async () => {
    const { container } = render(Assignees, {
      users: [
        {
          firstName: 'TEST',
          lastName: 'USER',
          email: 'test-user@bigcontent.io',
        },
      ],
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render multiple assignees', async () => {
    const { container } = render(Assignees, {
      users: [
        {
          firstName: 'TEST',
          lastName: 'USER',
          email: 'test-user@bigcontent.io',
        },
        {
          firstName: 'ANOTHER',
          lastName: 'USER',
          email: 'another-user@bigcontent.io',
        },
      ],
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not output any assignees if an empty array is passed', async () => {
    const { container } = render(Assignees, {
      users: [],
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not output any assignees if not passed any users', async () => {
    const { container } = render(Assignees);
    expect(container.firstChild).toMatchSnapshot();
  });
});
