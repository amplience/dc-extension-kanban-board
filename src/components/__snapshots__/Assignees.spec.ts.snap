// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assignees should not output any assignees if an empty array is passed 1`] = `
<div>
  <div
    class="assignees svelte-1ilxe5k"
    title=""
  />
</div>
`;

exports[`Assignees should not output any assignees if not passed any users 1`] = `
<div>
  <div
    class="assignees svelte-1ilxe5k"
    title=""
  />
</div>
`;

exports[`Assignees should render a single assignee 1`] = `
<div>
  <div
    class="assignees svelte-1ilxe5k"
    title="Assigned to TEST USER"
  >
    <div
      class="gravatar svelte-17a39iv"
    />
     
    
  </div>
</div>
`;

exports[`Assignees should render multiple assignees 1`] = `
<div>
  <div
    class="assignees svelte-1ilxe5k"
    title="Assigned to TEST USER, ANOTHER USER"
  >
    <div
      class="gravatar svelte-17a39iv"
    />
     
    <span
      class="svelte-1ilxe5k"
    >
      +
      1
    </span>
    
  </div>
</div>
`;
